File: python/Day_1/Question_19.py
number = int(input('enter value: '))

result = str(number)

count = 0

for r in result:
    if(len(result) == 2):
        temp = int(r) ** 2
        count += temp

    elif(len(result) == 3):
        temp = int(r) ** 3
        count += temp
    
    elif(len(result) == 4):
        temp = int(r) ** 4
        count += temp
    


if(number == count):
    print(f"{number} is Armstrong number.")
else:
    print(f"{number} is not Armstrong number.")
    

File: dsa/Day_4/while_loop/Question_19.cpp
#include <iostream>
#include <cmath>
using namespace std;

bool isArmstrong(int num) {  
    int originalNumber = num, reminder, result = 0, n = 0;

    int temp = num;
    while (temp != 0) {
        temp /= 10;
        n++;
    }

    temp = num;
    while (temp != 0) {
        reminder = temp % 10;
        result += pow(reminder, n);
        temp /= 10;
    }

    return (result == originalNumber);  
}

int main() {
    int n, i = 1;  

    cout << "Enter n: ";
    cin >> n;

    cout << "Armstrong numbers between 1 and " << n << " are: " << endl;
    while (i <= n) {
        if (isArmstrong(i)) {  
            cout << i << " ";
        }
        i++;
    }
    cout << endl;
    
    return 0;
}
